pharmaverse/aNCA
# Enhancement: Column Mapping Feature

## Issue

Closes #112 

## Description

We would like to offer the user better possibilities to define the column names that correspond with each column used by the app in the data. This will increase substantially the data flexbility and will also provide the user a better understanding of the App functioning

## Definition of Done

The requirements for the feature to be complete:
- [x] Provide select/selectize inputs for the user to define each of the important columns. Preferably if the layout is nice and with easy searchable options. One proposal is divided in these groups:

- Group identifiers. `Subject ID`, `Dose Number`, `Additional grouping`
- Concentration variables. `Analyte`, `Analyte value` 
- Dose variables. `Drug`, `Dose`, `Administration route`
- Time variables. `Time from 1st dose`, `Time from last dose`, `Dose duration`
- Unit variables. `Concentration unit`, `Time unit`

- [x] Hovering displays information about the column requirements or details. For example for administration route (`administration preferably stating either "intravascular" or "extravascular"`)

- [x] Provide inputs for the user to define the columns for unit measures (e.g, AVALU)

-  Formulas for PKNCAdata and PKNCAdose need to include `grouping_variables`

## How to test

Data Section -> Column mapping
Try using dataset with renamed columns to see that it still works.

## Contributor checklist
